File: Assets/Code/Outfit7/Raven/TrackComponents/AnimationData/CurveSpecialization/RavenAnimationDataCurveFloat.cs
//{ "engine_register_objects":["Starlite__Raven__RavenAnimationDataCurveFloat"] }
#if !__cplusplus
#pragma warning disable
namespace Starlite {

    namespace Raven {
    
        // RavenAnimationDataCurveFloat
        [Starlite.ObjectClassAttribute("Starlite::Raven::RavenAnimationDataCurveFloat")]
        public sealed partial class RavenAnimationDataCurveFloat : global::Starlite.Raven.RavenAnimationDataCurve<float> {
        }
        
    }
    
}
#pragma warning restore
#else
// RavenAnimationDataCurveFloat
namespace Starlite { namespace Raven { 
SIMPLEMENT_CLASS_WITH_PROPERTIES_IN_CLASS_SCOPED(Starlite::Raven::RavenAnimationDataCurveFloat, RavenAnimationDataCurveFloat, RavenAnimationDataCurveFloat, Starlite::Raven::RavenAnimationDataComponentBase, Starlite::SceneObjectComponent, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveFloat, m_Curves, m_Curves, kArray, kObject, Starlite::AnimationCurve::TypeId, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveFloat, m_EaseType, m_EaseType, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveFloat, m_ValueType, m_ValueType, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveFloat, m_RepeatCount, m_RepeatCount, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveFloat, m_Mirror, m_Mirror, kBool, kUnknown, -1, "")
SIMPLEMENT_CLASS_END(RavenAnimationDataCurveFloat)
} } 
SIMPLEMENT_CLASS_SCOPE(Starlite__Raven__RavenAnimationDataCurveFloat, Starlite::Raven, RavenAnimationDataCurveFloat)
#endif

File: Assets/Code/Outfit7/Raven/TrackComponents/AnimationData/CurveSpecialization/RavenAnimationDataCurveFloat.cc
#ifdef STARLITE
#include "RavenAnimationDataCurveFloat.h"
#include "RavenAnimationDataCurveFloat.cs"

namespace Starlite {
    namespace Raven {
        RavenAnimationDataCurveFloat::RavenAnimationDataCurveFloat() {
        }

        float RavenAnimationDataCurveFloat::EvaluateAtTime(double time, double duration) {
            return m_Curves[0]->Evaluate((float)GetNormalizedTime(GetCurrentTime(time, duration), duration, m_EaseType));
        }

        void RavenAnimationDataCurveFloat::SetStartingValues(const float& values) {
            m_Curves = Array<Ref<AnimationCurve>>(1);
            m_Curves[0] = Object::New<AnimationCurve>();
            m_Curves[0]->AddKeyframe(AnimationCurve::Keyframe(0, values));
            m_Curves[0]->AddKeyframe(AnimationCurve::Keyframe(1, values));
        }

        void RavenAnimationDataCurveFloat::SyncStartingValues(const float& values) {
            auto& curve = m_Curves[0];
            auto& key = curve->GetKeyframes()[0];
            key.value = values;
        }
    } // namespace Raven
} // namespace Starlite
#endif